aea/manager/project.py

```diff
@@ -0,0 +1,246 @@
+
+    @property
+    def builder(self) -> AEABuilder:
+        """Get builder instance."""
+        builder = self._get_builder(self.agent_config, self.project.path)
+        if not builder.private_key_paths:
+            # no keys, but create one for builder only
+            default_ledger = builder.get_default_ledger()
+            builder.add_private_key(
+                default_ledger, self._create_private_key(default_ledger)
+            )
+        builder.set_name(self.agent_name)
+        builder.set_runtime_mode("threaded")
+        return builder
+
+    @property
+    def agent_config(self) -> AgentConfig:
+        """Get agent config."""
+        return self._agent_config
+
+    def _create_private_key(self, ledger: str, replace: bool = False) -> str:
+        """
+        Create new key for agent alias in working dir keys dir.
+
+        If file exists, check `replace` option.
+        """
+        filepath = os.path.join(
+            self._keys_dir, f"{self.agent_name}_{ledger}_private.key"
+        )
+        if os.path.exists(filepath) and not replace:
+            return filepath
+        create_private_key(ledger, filepath)
+        return filepath
+
+    def remove_from_project(self):
+        """Remove agent alias from project."""
+        self.project.agents.remove(self.agent_name)
+
+    @property
+    def dict(self) -> Dict[str, Any]:
+        """Convert AgentAlias to dict."""
+        return {
+            "public_id": str(self.project.public_id),
+            "agent_name": self.agent_name,
+            "config": self.config_json,
+        }
+
+    @property
+    def config_json(self) -> List[Dict]:
+        """Get agent config json data."""
+        json_data = self.agent_config.ordered_json
+        result: List[Dict] = [json_data] + json_data.pop("component_configurations", {})
+        return result
+
+    def get_aea_instance(self) -> AEA:
+        """Build new aea instance."""
+        return self.builder.build()
+
+    def set_overrides(
+        self,
+        agent_overrides: Optional[Dict] = None,
+        component_overrides: Optional[List[Dict]] = None,
+    ) -> None:
+        """Set override for this agent alias's config."""
+        overrides = deepcopy(agent_overrides or {})
+        component_configurations: Dict[ComponentId, Dict] = {}
+
+        for component_override in deepcopy(component_overrides or []):
+            component_id = ComponentId.from_json(
+                {"version": "any", **component_override}
+            )
+            component_override.pop("author")
+            component_override.pop("name")
+            component_override.pop("type")
+            component_override.pop("version")
+            component_configurations[component_id] = component_override
+
+        overrides["component_configurations"] = component_configurations
+        return self.agent_config_manager.update_config(overrides)
```

Comment:
this could fail..provide proper error message

Reply:
done, checks for Value and Key errors